pkg/provisioner/ironic/ironic.go

```diff
@@ -609,6 +609,57 @@ func (p *ironicProvisioner) UpdateHardwareState() (result provisioner.Result, er
 	return result, nil
 }
 
+func (p *ironicProvisioner) startManualCleaning(ironicNode *nodes.Node) (result provisioner.Result, err error) {
+	var cleanSteps []nodes.CleanStep
+
+	if p.bmcAccess.RAIDInterface() != "" {
+		// Set raid config
+		err = setTargetRAIDCfg(p.client, ironicNode, p.host.Status.Provisioning.RAID)
+		if err != nil {
+			p.log.Error(err, "failed in setting the target RAID configuration")
+			return
+		}
+		// Build raid configuration clean steps
+		cleanSteps = append(cleanSteps, buildRAIDCleanSteps(p.host.Status.Provisioning.RAID)...)
+	}
```

Comment:
else? should we raise if the node does not support RAID (e.g. an IPMI node)?

Reply:
With the error handling changes @zaneb and @andfasano are working on, if we report an error we will just keep retrying the provisioning operation. Is that what we want?

Reply:
How do we resolve it? Do we bail out completely? or proceed without cleaning?

Reply:
If the node doesn't support RAID, we can check `p.host.Status.Provisioning.RAID`'s value, if nil output a message, if not nil return an error.

```go
else {
	if p.host.Status.Provisioning.RAID != nil {
		p.log.Error("the BMC of the current node does not support RAID", "BMC", p.bmcAccess.Driver())
		return result, fmt.Errorf("the BMC of the current node does not support RAID")
	}
	p.log.Info("the BMC of the current node does not support RAID", "BMC", p.bmcAccess.Driver())
}
```